shared/lib_utility_rate.cpp

```diff
@@ -143,7 +143,7 @@ UtilityRateForecast::UtilityRateForecast(rate_data* util_rate, size_t stepsPerHo
 	rate = std::shared_ptr<rate_data>(new rate_data(*util_rate));
 	m_monthly_load_forecast = monthly_load_forecast;
 	m_monthly_gen_forecast = monthly_gen_forecast;
-	m_monthly_peak_forecast = monthly_peak_forecast;
+	m_monthly_avg_load_forecast = monthly_avg_load_forecast;
```

Comment:
While I see how going from max net peak to average gross load for the forecast's baseline assumption for demand makes sense, does the average gross load actually work better than max gross peak?

Also, in all the tests the input to the constructor for the forecast uses the name `monthly_peak_forecast` instead of `monthly_avg_load_forecast`.

Depending on the shape of the load, the average gross load can be very different from the max gross peak (imagine a single peak vs a uniform load).

Reply:
The m_monthly_peak_forecast was never used, and I neglected to remove it in my final code cleanup. I went through and renamed the variables in the code for this update, I'll rename the tests shortly. The previous code was using average net.

The purpose of that line is to prevent the first hour(s) of each month from looking like a peak if they are below the average load. I'm not sure what the algorithm would do with max gross or max net. I could imagine setting this variable to (max net - battery power) but that means the price forecast would need to be aware of the battery power, which seems convoluted. Additionally, that value could end up being lower than the typical nightly load - the case I was working with the max net peak was 625 kW, the battery power was 200 kW, and the typical nightly load was 440 kW, so there would still be some degree of predicted demand charge on the first hour of each month with that approach. Average gross load was ~490 kW, and average net was ~350 kW.

Reply:
Okay it sounds like the baseline you're looking for in this case is the some lower estimate of demand peak after battery dispatch-- therefore you can compare how much different dispatch plans __increase__ the demand peak. In my mind, I was thinking of the baseline being the original demand peak, whereby you would compare how much different dispatch plans __decrease__ the demand peak. IIRC (but haven't looked at this in some time) that's similar to what the peak shaving algorithm does in sorting the peaks by highest existing peak.